.github/workflows/check-catalyst.yaml

```diff
@@ -234,6 +242,15 @@ jobs:
         enableCrossOsArchive: True
         fail-on-cache-miss: True
 
+    - name: Get Cached MHLO Source
+      id: cache-mhlo-source
```

Comment:
I think some header files are not copied into mhlo build. And mhlo build is also needed due to some auto generated files. I tested this by deleting mhlo source ^ mhlo build and trying to build, and I was unable to. This is, I think, because our CMake script for making the dialects does not build MHLO. Instead uses the MHLO build directory to locate the mhlo libraries `${ALL_MHLO_PASSES}`. We could in theory build these libraries here as opposed to a different CI/CD step.

Reply:
We will also need it for scattering lowering btw